common/unit_test/Test_Padding.cpp

```diff
@@ -43,434 +65,535 @@ void test_reshape1D_1DView(bool is_C2R) {
+  View6D<T> x_out("x_out", n0, n1, n2, n3, n4, n5);
   constexpr int DIM = 6;
   shape_type<6> default_shape({n0, n1, n2, n3, n4, n5});
 
   for (int axis0 = 0; axis0 < DIM; axis0++) {
     for (int axis1 = 0; axis1 < DIM; axis1++) {
       if (axis0 == axis1) continue;
       axes_type<2> axes({axis0, axis1});
+      shape_type<6> in_shape = default_shape;
+      in_shape.at(axis1)     = get_c2r_shape(x_out.extent(axis1), is_C2R);
+      auto [_n0, _n1, _n2, _n3, _n4, _n5] = in_shape;
+      View6D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2, _n3, _n4,
+                                           _n5);
       for (int i0 = -1; i0 <= 1; i0++) {
         for (int i1 = -1; i1 <= 1; i1++) {
           shape_type<6> ref_shape = default_shape;
-          std::size_t n0_new  = static_cast<std::size_t>(x.extent(axis0) + i0);
-          std::size_t n1_new  = static_cast<std::size_t>(x.extent(axis1) + i1);
+          std::size_t n0_new =
+              static_cast<std::size_t>(x_in.extent(axis0) + i0);
+          std::size_t n1_new =
+              static_cast<std::size_t>(x_in.extent(axis1) + i1);
           ref_shape.at(axis0) = n0_new;
           ref_shape.at(axis1) = get_c2r_shape(n1_new, is_C2R);
 
           shape_type<2> new_shape = {n0_new, n1_new};
 
           auto modified_shape =
-              KokkosFFT::Impl::get_modified_shape(x, new_shape, axes, is_C2R);
+              KokkosFFT::Impl::get_modified_shape(x_in, x_out, new_shape, axes);
           EXPECT_TRUE(modified_shape == ref_shape);
         }
       }
     }
   }
 }
 
-void test_reshape2D_7DView(bool is_C2R) {
+template <typename T>
+void test_reshape2D_7DView() {
   const int n0 = 5, n1 = 11, n2 = 10, n3 = 8, n4 = 3, n5 = 4, n6 = 7;
-  View7D<double> x("x", n0, n1, n2, n3, n4, n5, n6);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+  View7D<T> x_out("x_out", n0, n1, n2, n3, n4, n5, n6);
   constexpr int DIM = 7;
   shape_type<7> default_shape({n0, n1, n2, n3, n4, n5, n6});
 
   for (int axis0 = 0; axis0 < DIM; axis0++) {
     for (int axis1 = 0; axis1 < DIM; axis1++) {
       if (axis0 == axis1) continue;
       axes_type<2> axes({axis0, axis1});
+      shape_type<7> in_shape = default_shape;
+      in_shape.at(axis1)     = get_c2r_shape(x_out.extent(axis1), is_C2R);
+      auto [_n0, _n1, _n2, _n3, _n4, _n5, _n6] = in_shape;
+      View7D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2, _n3, _n4, _n5,
+                                           _n6);
       for (int i0 = -1; i0 <= 1; i0++) {
         for (int i1 = -1; i1 <= 1; i1++) {
           shape_type<7> ref_shape = default_shape;
-          std::size_t n0_new  = static_cast<std::size_t>(x.extent(axis0) + i0);
-          std::size_t n1_new  = static_cast<std::size_t>(x.extent(axis1) + i1);
+          std::size_t n0_new =
+              static_cast<std::size_t>(x_in.extent(axis0) + i0);
+          std::size_t n1_new =
+              static_cast<std::size_t>(x_in.extent(axis1) + i1);
```

Comment:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.